### DISCUSSION TITLE
Sync `cwd` to current buffer

### DISCUSSION THREAD

**Discussion by altschuler** [helix-editor/helix#4200] [Ideas] [answered]
I've read https://github.com/helix-editor/helix/pull/3523 and https://github.com/helix-editor/helix/discussions/2481, but I'm suggestion a feature that automatically changes the cwd to the current buffers path, certainly something you could turn on/off. For me personally, I find it easier to work with files and commands when the current buffer determines the cwd. 

For instance, I'm working in a monorepo where each sub-project has their own formatting defined (eg there's a formatter in `/project-a/formatting.json`, and `/project-b/formatting.json`. Currently I have to `:cd project-a` and `:cd project-b` all the time before saving because the formatter otherwise reads the wrong config. 

Also when creating and opening files I find it easier to figure out when I'm starting from the current file, rather than having to drill deep into the project.

**Comment by the-mikedavis** [comment, marked as answer]
This would be covered by https://github.com/helix-editor/helix/issues/2056

**Reply by altschuler** [reply]
Oh, I hadn't seen that one. Thanks :)
